package/MDAnalysis/coordinates/XTC.py

```diff
@@ -1,75 +1,123 @@
-
-The Gromacs `XTC trajectory format`_ is a format with lossy
-compression. Coordinates are only stored with a fixed precision (by
-default, 1/1000 of a nm). The XTC format can only store
-*coordinates*. Its main advantage is that it requires less disk space
-than e.g. TRR or DCD trajectories and the loss of precision is usually
-not a problem.
-
-If one wants to store Gromacs trajectories without loss of precision
-or with velocities and/or forces then one should use the TRR format
-(see module :mod:`~MDAnalysis.coordinates.TRR`).
-
-The XTC I/O interface uses
-:mod:`~MDAnalysis.coordinates.xdrfile.libxdrfile2` to implement random
-access to frames. This works by initially building an internal index
-of all frames and then using this index for direct seeks. Building the
-index is triggered by
-:func:`~MDAnalysis.coordinates.xdrfile.libxdrfile2.read_xtc_n_frames`,
-which typically happens when one accesses the
-:attr:`XTCReader.n_frames` attribute for the first time. Building the
-index may take many minutes for large trajectories but afterwards
-access is faster than with native Gromacs tools.
-
-.. _XTC trajectory format:
-   http://www.gromacs.org/Documentation/File_Formats/.xtc_File
-
-.. versionchanged:: 0.8.0
-   The XTC I/O interface now uses
-   :mod:`~MDAnalysis.coordinates.xdrfile.libxdrfile2`, which has
-   seeking and indexing capabilities. Note that unlike
-   :mod:`~MDAnalysis.coordinates.xdrfile.libxdrfile` before it,
-   :mod:`~MDAnalysis.coordinates.xdrfile.libxdrfile2` is distributed
-   under the GNU GENERAL PUBLIC LICENSE, version 2 (or higher).
-   :class:`~MDAnalysis.coordinates.XTC.Timestep` now correctly
-   deals with presence/absence of coordinate/velocity/force
-   information on a per-frame basis.
-
-Module reference
-----------------
-
-.. autoclass:: Timestep
-   :members:
-   :inherited-members:
-
-.. autoclass:: XTCReader
-   :members:
-   :inherited-members:
-
-.. autoclass:: XTCWriter
-   :members:
-   :inherited-members:
-
-"""
-
-from .xdrfile.XTC import XTCReader, XTCWriter, Timestep
+import errno
+
+import numpy as np
+
+from . import base
+from ..lib.formats.xtc import XTCFile
+from ..lib.mdamath import triclinic_box
+from ..core.AtomGroup import Universe, AtomGroup
+
+
+class XTCReader(base.Reader):
+    format = 'XTC'
+    units = {'time': 'ps', 'length': 'nm'}
+
+    def __init__(self, filename, convert_units=True, sub=None, **kwargs):
+        super(XTCReader, self).__init__(filename, convert_units=convert_units,
+                                        **kwargs)
+        self.xtc = XTCFile(filename)
+
+        self.n_frames = len(self.xtc)
+        xyz, box, step, time, prec = self.xtc.read()
+        try:
+            xtc_frame = self.xtc.read()
+            time_2 = xtc_frame[3]
+        except:
+            time_2 = 2 * time
+        self._sub = sub
+        if sub is not None:
+            self.n_atoms = len(sub)
+        else:
+            self.n_atoms = self.xtc.n_atoms
+        self.xtc.seek(0)
```

Comment:
Is this going to put the trajectory before frame 0?  The Reader should read the first frame, so that calling next reads the second frame

```python
u = mda.Universe(etc)

u.trajectory.next()  # should be second frame

```